prereise/gather/demanddata/bldg_electrification/floor_area_avg_slopes.py

```diff
@@ -408,6 +421,101 @@ def main_plots(iso, zone_shape, pumas_shp, state_shp, plot_show=True):
             zone_shp_ri,
         ]:
             zone_shp = zone_shp.append(i)
+    elif iso == "PJM":
+        for i in [
+            zone_shp_de,
+            zone_shp_il,
+            zone_shp_in,
+            zone_shp_ky,
+            zone_shp_md,
+            zone_shp_nc,
+            zone_shp_mi,
+            zone_shp_nj,
+            zone_shp_oh,
+            zone_shp_pa,
+            zone_shp_va,
+            zone_shp_wva,
+            zone_shp_tn,
+            zone_shp_dc,
+        ]:
+            zone_shp = zone_shp.append(i)
+    elif iso == "SPP":
+        for i in [
+            zone_shp_ks,
+            zone_shp_ok,
+            zone_shp_nm,
+            zone_shp_tx,
+            zone_shp_ar,
+            zone_shp_la,
+            zone_shp_mo,
+            zone_shp_sd,
+            zone_shp_nd,
+            zone_shp_mt,
+            zone_shp_mn,
+            zone_shp_ia,
+            zone_shp_wy,
+            zone_shp_ne,
+        ]:
+            zone_shp = zone_shp.append(i)
+    elif iso == "MISO":
+        for i in [
+            zone_shp_la,
+            zone_shp_ar,
+            zone_shp_ms,
+            zone_shp_mi,
+            zone_shp_mo,
+            zone_shp_ky,
+            zone_shp_in,
+            zone_shp_il,
+            zone_shp_ia,
+            zone_shp_mn,
+            zone_shp_wi,
+            zone_shp_nd,
+            zone_shp_sd,
+            zone_shp_tx,
+            zone_shp_mt,
+        ]:
+            zone_shp = zone_shp.append(i)
+    elif iso == "SW":
+        for i in [
+            zone_shp_az,
+            zone_shp_nm,
+            zone_shp_co,
+            zone_shp_nv,
+            zone_shp_wy,
+            zone_shp_sd,
+            zone_shp_ne,
+        ]:
+            zone_shp = zone_shp.append(i)
+    elif iso == "NW":
+        for i in [
+            zone_shp_ca,
+            zone_shp_wa,
+            zone_shp_or,
+            zone_shp_id,
+            zone_shp_nv,
+            zone_shp_ut,
+            zone_shp_wy,
+            zone_shp_mt,
+        ]:
+            zone_shp = zone_shp.append(i)
+    elif iso == "SE":
+        for i in [
+            zone_shp_mo,
+            zone_shp_ky,
+            zone_shp_ms,
+            zone_shp_tn,
+            zone_shp_al,
+            zone_shp_ga,
+            zone_shp_nc,
+            zone_shp_sc,
+            zone_shp_fl,
+            zone_shp_va,
```

Comment:
Another option is defining these mappings in a dictionary, such as `iso_state_overlay = {"NE": [zone_shp_ma, ...], "PJM":[...] ...}` then later we can simply do `zone_shp = pd.concat([s for s in iso_state_overlay[iso])`

Reply:
Made the changes